server/src/main/java/org/tctalent/server/service/db/audit/Auditor.java

```diff
@@ -31,121 +36,98 @@
 import org.tctalent.server.repository.db.AuditLogRepository;
 import org.tctalent.server.security.AuthService;
 
-import java.time.OffsetDateTime;
-import java.util.List;
-import java.util.regex.Matcher;
-import java.util.regex.Pattern;
-import java.util.stream.Collectors;
-
 @Aspect
 @Component
 public class Auditor {
 
-    private static final Logger log = LoggerFactory.getLogger(Auditor.class);
-
-    private final AuditLogRepository auditLogRepository;
-    private final AuthService authService;
-    public static final Long ANONYMOUS_USER_ID = -999L;
-
+  private static final Logger log = LoggerFactory.getLogger(Auditor.class);
+
+  private final AuditLogRepository auditLogRepository;
+  private final AuthService authService;
+  public static final Long ANONYMOUS_USER_ID = -999L;
+
+
+  public Auditor(AuditLogRepository auditLogRepository, AuthService authService) {
+    this.auditLogRepository = auditLogRepository;
+    this.authService = authService;
+  }
+
+  @AfterReturning(
+      pointcut = "@annotation(audit) && args(inputValue,..)",
+      returning = "returnValue")
+  public void handleAuditLog(Audit audit, Object inputValue, Object returnValue) {
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

The cyclomatic complexity of this method is 17. By comparison, 99% of the methods in the CodeGuru reference dataset have a lower cyclomatic complexity. This indicates the method has a high number of decisions and it can make the logic difficult to understand and test. We recommend that you simplify this method or break it into multiple methods.